### PR TITLE
Endpoint implementation to get list of users blocked by a user

### LINKED ISSUE #154: Endpoint implementation to get list of users blocked by a user
**Describe the task that needs to be done.**

Implement a new API endpoint for getting a list of users blocked by a user.

**Describe how a solution to your proposed task might look like (and any alternatives considered).**

Using the URL path `/block` with a HTTP GET request, rows from the `BLOCKED_BUDDIES` database table can be selected in queries where `USER_BLOCKER_ID` matches `userId` for getting a list of user's blocked by `userId`. This can be returned in the HTTP response entity as a list of `UserDTO`.

**Notes**

### PR DESCRIPTION
**Pull request by R055A**
## Description

Solves Issue #154 

Implement endpoint functionality for getting a list of users blocked by a user from the BLOCKED_BUDDIES database table and returning this in the response JSON for a HTTP GET request to `/api/users/block`.

## How has this been tested?

Automated testing.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/main/java/com/team701/buddymatcher/controllers/users/UserController.java [hunk 1]
```diff
@@ -275,4 +275,21 @@ public UserController(UserService userService, TimetableService timetableService
...
         }
     }
+
+    @Operation(summary = "Get method for getting a list of users blocked by a user")
+    @GetMapping(path = "/block", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<List<UserDTO>> getBlockedBuddies(@Parameter(hidden = true)
+                                                           @SessionAttribute("UserId") Long userBlockingId) {
+        try {
+            List<User> blockedUsers = userService.getBlockedBuddies(userBlockingId);
+            System.out.println(blockedUsers);
```
Comment (kimslor) [inline src/main/java/com/team701/buddymatcher/controllers/users/UserController.java]:
```suggestion
```

File: src/main/java/com/team701/buddymatcher/controllers/users/UserController.java [hunk 2]
```diff
@@ -275,4 +275,21 @@ public UserController(UserService userService, TimetableService timetableService
...
+            List<UserDTO> userDTOs = blockedUsers.stream()
+                    .map(user -> modelMapper.map(user, UserDTO.class))
+                    .collect(Collectors.toList());
+            System.out.println(userDTOs);
```
Comment (kimslor) [inline src/main/java/com/team701/buddymatcher/controllers/users/UserController.java]:
```suggestion
```

**Review by kimslor — CHANGES_REQUESTED** [review]

File: src/main/java/com/team701/buddymatcher/controllers/users/UserController.java (hunk 2 above)
Comment (r055a) [reply src/main/java/com/team701/buddymatcher/controllers/users/UserController.java]: Thank you for spotting that!

File: src/main/java/com/team701/buddymatcher/controllers/users/UserController.java (hunk 1 above)
Comment (r055a) [reply src/main/java/com/team701/buddymatcher/controllers/users/UserController.java]: Thank you for spotting that!

**Review by kimslor — APPROVED** [review]

**Pull request merged**